xplique/concepts/craft.py

```diff
@@ -0,0 +1,944 @@
+"""
+CRAFT Module for Tensorflow/Pytorch
+"""
+
+from abc import ABC, abstractmethod
+import dataclasses
+from enum import Enum
+import colorsys
+from math import ceil
+
+import numpy as np
+import cv2
+from sklearn.decomposition import NMF
+from sklearn.exceptions import NotFittedError
+import matplotlib.pyplot as plt
+from matplotlib.colors import ListedColormap, LinearSegmentedColormap
+from matplotlib import gridspec
+from tqdm import tqdm
+
+from xplique.attributions.global_sensitivity_analysis import (HaltonSequenceRS, JansenEstimator)
+
+from ..types import Callable, Tuple, Optional
+from .base import BaseConceptExtractor
+
+@dataclasses.dataclass
+class Factorization:
+    """ Dataclass handling data produced during the Factorization step."""
+    inputs: np.ndarray
+    class_id: int
+    crops: np.ndarray
+    reducer: NMF
+    crops_u: np.ndarray
+    concept_bank_w: np.ndarray
+
+class Sensitivity:
+    """ Dataclass handling data produced during the Sobol indices computation. """
+
+    def __init__(self, importances: np.ndarray, most_important_concepts: np.ndarray,
+                 cmaps: list=None):
+        self.importances = importances
+        self.most_important_concepts = most_important_concepts
+        self.set_concept_attribution_cmap(cmaps=cmaps)
+
+    @staticmethod
+    def _get_alpha_cmap(cmap):
```

Comment:
It would be nice to document this class as it is not straightforward how it will be used :)

Reply:
done